OneSignalSDK/onesignal/src/main/java/com/onesignal/GenerateNotification.java

```diff
@@ -133,28 +122,20 @@ private static CharSequence getTitle(JSONObject fcmJson) {
    }
 
    private static PendingIntent getNewActionPendingIntent(int requestCode, Intent intent) {
-      if (openerIsBroadcast)
-         return PendingIntent.getBroadcast(currentContext, requestCode, intent, PendingIntent.FLAG_UPDATE_CURRENT);
       return PendingIntent.getActivity(currentContext, requestCode, intent, PendingIntent.FLAG_UPDATE_CURRENT);
    }
 
    private static Intent getNewBaseIntent(int notificationId) {
-      Intent intent = new Intent(currentContext, notificationOpenedClass)
-                        .putExtra(BUNDLE_KEY_ANDROID_NOTIFICATION_ID, notificationId);
-
-      if (openerIsBroadcast)
-         return intent;
-      return intent.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP | Intent.FLAG_ACTIVITY_CLEAR_TOP);
+     return new Intent(currentContext, notificationOpenedClass)
+             .putExtra(BUNDLE_KEY_ANDROID_NOTIFICATION_ID, notificationId)
+             .addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP | Intent.FLAG_ACTIVITY_CLEAR_TOP);
    }
 
    private static Intent getNewBaseDeleteIntent(int notificationId) {
-      Intent intent = new Intent(currentContext, notificationOpenedClass)
-          .putExtra(BUNDLE_KEY_ANDROID_NOTIFICATION_ID, notificationId)
-          .putExtra("dismissed", true);
-
-      if (openerIsBroadcast)
-         return intent;
-      return intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_MULTIPLE_TASK | Intent.FLAG_ACTIVITY_NO_ANIMATION);
+      return new Intent(currentContext, notificationOpenedClass)
+              .putExtra(BUNDLE_KEY_ANDROID_NOTIFICATION_ID, notificationId)
+              .putExtra("dismissed", true)
+              .addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_MULTIPLE_TASK | Intent.FLAG_ACTIVITY_NO_ANIMATION);
```

Comment:
Mike tried to switch this notification dismiss action to an `Activity` before in the past and it resulted in a visual flicker. Do these flags completely prevent that?

Reply:
good catch! there is no flicker but the process is being closed

Reply:
We might have to use a `BroadcastReceiver` for the dismisses. If we use an `Activity` for dismisses it could result in odd focus behavior in the app with extra onPause and onResume calls each time a notification is swiped away we would want to avoid.

Reply:
Yeah, seems like the most reasonable solution, using intent for dismissing tracking, ends on weirds behaviors